src/expressions/include/antares/expressions/visitors/EvaluationContext.h

```diff
@@ -44,18 +61,34 @@ class EvaluationContext
      * @return The value of the parameter.
      * @throws std::out_of_range If the parameter is not found.
      */
-    double getParameterValue(const std::string& key) const;
+    double getSystemParameterValueAsDouble(const std::string& key) const;
+    std::string getSystemParameterValue(const std::string& key) const;
+
+    double getParameterValue(const std::string& key,
+                             const std::string& scenarioGroup,
+                             const unsigned scenario,
+                             unsigned int hour) const;
+
+    ParameterType getParameterType(const std::string& key) const;
+    ContextParameter getParameter(const std::string& key) const;
+
+    template<class T>
+    struct CouldNotEvaluateConstantParameter: T
+    {
+        using T::T;
+    };
 
 private:
     /**
      * @brief A map storing parameter values.
      */
-    std::map<std::string, double> parameters_;
+    std::map<std::string, ContextParameter> system_parameters_;
```

Comment:
this class isn't supposed to know what a system is. maybe some name like this?
```suggestion
    std::map<std::string, ContextParameter> parameters_types_and_values_;
```

Reply:
done